File: LinkCollector-msSQL/LinkCollector-msSQL.OB/CLinkCategory.vb
Imports BookmarSorter.DL

Public Class CLinkCategory
    Private mstrName As String
    Private mstrFileName As String
    Private mstrLinkDefine As String
    Dim msFileName As String = "LinkCategories.txt"
    Public Property Name() As String
        Get
            Return mstrName
        End Get
        Set(ByVal value As String)
            mstrName = value
        End Set
    End Property
    Public Property FileName() As String
        Get
            Return mstrFileName
        End Get
        Set(ByVal value As String)
            mstrFileName = value
        End Set
    End Property
    Public Property LinkDefine() As String
        Get
            Return mstrLinkDefine
        End Get
        Set(ByVal value As String)
            mstrLinkDefine = value
        End Set
    End Property
    Public Sub New(ByVal vsName As String, ByVal vsFileName As String, ByVal vsLinkDefine As String)
        mstrName = vsName
        mstrFileName = vsFileName
        mstrLinkDefine = vsLinkDefine
    End Sub
    Public Sub New()
        mstrName = ""
        mstrFileName = ""
        mstrLinkDefine = ""
    End Sub
    Public Function Save() As Boolean

        Try

            Dim oFiler As New CFile()
            oFiler.Save(mstrName & "|" & mstrFileName & "|" & mstrLinkDefine)
            Return True

        Catch ex As Exception

            Throw ex


        End Try


    End Function
    Public Function Save(ByVal vsFileName As String) As Boolean

        Try

            Dim oFiler As New CFile(vsFileName)
            oFiler.Save(mstrName & "|" & mstrFileName & "|" & mstrLinkDefine)
            Return True

        Catch ex As Exception

            Throw ex


        End Try


    End Function
End Class

File: LinkListBuilder/FFTool.OB/cLinkCategory.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Utilities.DL;
namespace FFTool.OB
{
	public class cLinkCategory
	{
		private string mstrName;
	private string mstrFileName;
	private string mstrLinkDefine;
	string msFileName = "LinkCategories.txt";
	public string Name {
		get { return mstrName; }
		set { mstrName = value; }
	}
	public string FileName {
		get { return mstrFileName; }
		set { mstrFileName = value; }
	}
	public string LinkDefine {
		get { return mstrLinkDefine; }
		set { mstrLinkDefine = value; }
	}
	public cLinkCategory(string vsName, string vsFileName, string vsLinkDefine)
	{
		mstrName = vsName;
		mstrFileName = vsFileName;
		mstrLinkDefine = vsLinkDefine;
	}
	public cLinkCategory()
	{
		mstrName = "";
		mstrFileName = "";
		mstrLinkDefine = "";
	}
	public bool Save()
	{


		try {
			cFile oFiler = new cFile();
			oFiler.Write(mstrName + "|" + mstrFileName + "|" + mstrLinkDefine);
			return true;


		} catch (Exception ex) {
			throw ex;


		}


	}
	public bool Save(string vsFileName)
	{


		try {
			cFile oFiler = new cFile(vsFileName);
			oFiler.Write(mstrName + "|" + mstrFileName + "|" + mstrLinkDefine);
			return true;


		} catch (Exception ex) {
			throw ex;


		}


	}
	}
}
